src/platform/platform_linux.cpp

```diff
@@ -38,32 +38,103 @@
 #include "shared/sshfs_server_process_spec.h"
 #include <disabled_update_prompt.h>
 
+#include <QDir>
+#include <QFile>
+#include <QRegularExpression>
+#include <QString>
+#include <QTextStream>
+
+#include <linux/if_arp.h>
+
 namespace mp = multipass;
 namespace mpl = multipass::logging;
 namespace mu = multipass::utils;
 
 namespace
 {
 constexpr auto autostart_filename = "multipass.gui.autostart.desktop";
+constexpr auto category = "Linux platform";
+
+// Fetch the ARP protocol HARDWARE identifier.
+int get_net_type(const QDir& net_dir) // types defined in if_arp.h
+{
+    static constexpr auto default_ret = -1;
+    static const auto type_filename = QStringLiteral("type");
+
+    QFile type_file{net_dir.filePath(type_filename)};
+    if (type_file.open(QIODevice::ReadOnly | QIODevice::Text))
+    {
+        bool ok;
+        auto got = QTextStream{&type_file}.read(6).toInt(&ok); // 6 chars enough for up to 0xFFFF; 0 returned on failure
+        return ok ? got : default_ret;
+    }
+
+    mpl::log(mpl::Level::debug, category, fmt::format("Could not read {}", type_file.fileName()));
+    return default_ret;
+}
+
+// device types found in Linux source (in drivers/net/): PHY, bareudp, bond, geneve, gtp, macsec, ppp, vxlan, wlan, wwan
+// should be empty for ethernet
+QString get_net_devtype(const QDir& net_dir)
+{
+    static constexpr auto max_read = 5000;
+    static const auto uevent_filename = QStringLiteral("uevent");
+    static const auto devtype_regex =
+        QRegularExpression{QStringLiteral("^DEVTYPE=(.*)$"), QRegularExpression::MultilineOption};
+
+    QFile uevent_file{net_dir.filePath(uevent_filename)};
+    if (uevent_file.open(QIODevice::ReadOnly | QIODevice::Text))
+    {
+        auto contents = QTextStream{&uevent_file}.read(max_read);
+        return devtype_regex.match(contents).captured(1);
+    }
+
+    mpl::log(mpl::Level::debug, category, fmt::format("Could not read {}", uevent_file.fileName()));
+    return {};
+}
+
+bool is_virtual_net(const QDir& net_dir)
+{
+    static const auto virtual_dir = QStringLiteral("virtual");
+
+    auto fixme = net_dir.canonicalPath();
+    return net_dir.canonicalPath().contains(virtual_dir, Qt::CaseInsensitive);
+}
+
+bool is_ether_net(const QDir& net_dir)
```

Comment:
```suggestion
bool is_ethernet(const QDir& net_dir)
```

I'd say it's an Ethernet Network, not an Ether Network :]

Reply:
I suppose that's what ethernet means originally. I went for that for consistency with the other ones, and following the name of the constant (`ARPHRD_ETHER`). It is abbreviated like that in other places too (e.g. [EtherType](https://en.wikipedia.org/wiki/EtherType)). But ok, I don't mind changing it.

BTW, here's a history quote from [wikipedia](https://en.wikipedia.org/wiki/Ethernet#History):

> The idea was first documented in a memo that Metcalfe wrote on May 22, 1973, where he named it after the luminiferous aether once postulated to exist as an "omnipresent, completely-passive medium for the propagation of electromagnetic waves."